### PR TITLE
14519 Update entity-filer to support Correction of ULC, CCC, BC

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* /bcgov/entity#14519

*Description of changes:*
- added ULC, CCC, BC to correction
- unit test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/filing_processors/correction.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def process(correction_filing: Filing, filing: Dict, filing_meta: FilingMeta, bu
             staff_id=correction_filing.submitter_id
         )
     )
-    if business.legal_type in ['SP', 'GP', 'BEN']:
+    if business.legal_type in ['SP', 'GP', 'BC', 'BEN', 'CCC', 'ULC']:
```
Comment (argush3) [inline queue_services/entity-filer/src/entity_filer/filing_processors/correction.py:51]: The legal type is actually `CC`.  It's confusing even though it's called CCC.

File: queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py [hunk 2]
```diff
@@ -407,49 +419,49 @@ async def test_worker_correction_court_order(app, session, mocker, test_name, le
 @pytest.mark.parametrize(
     'test_name',
     [
-        'ben_add_director',
-        'ben_edit_director_name_and_address',
-        'ben_delete_director',
+        'bc_add_director',
+        'bc_edit_director_name_and_address',
+        'bc_delete_director',
```
Comment (argush3) [inline queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py:424]: can we update this test to test all the relevant entity types?  i.e. BEN, BC, ULC, CC

File: queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py [hunk 3]
```diff
@@ -223,8 +223,14 @@
...
     'test_name, legal_name, new_legal_name, legal_type, filing_template',
     [
-        ('name_change', 'Test Firm', 'New Name', 'BEN', BEN_CORRECTION),
-        ('no_change', 'Test Firm', None, 'BEN', BEN_CORRECTION)
+        ('name_change', 'Test Firm', 'New Name', 'BC', BC_CORRECTION),
+        ('no_change', 'Test Firm', None, 'BC', BC_CORRECTION),
+        ('name_change', 'Test Firm', 'New Name', 'BEN', BC_CORRECTION),
+        ('no_change', 'Test Firm', None, 'BEN', BC_CORRECTION),
+        ('name_change', 'Test Firm', 'New Name', 'CCC', BC_CORRECTION),
+        ('no_change', 'Test Firm', None, 'CCC', BC_CORRECTION),
```
Comment (argush3) [inline queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py:231]: CCC should be CC

File: queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py [hunk 4]
```diff
@@ -355,7 +364,10 @@ async def test_correction_business_address(app, session, mocker, test_name, lega
 @pytest.mark.parametrize(
     'test_name, legal_type, filing_template',
     [
-        ('ben_court_order', 'BEN', BEN_CORRECTION),
+        ('bc_court_order', 'BC', BC_CORRECTION),
+        ('ben_court_order', 'BEN', BC_CORRECTION),
+        ('ccc_court_order', 'CCC', BC_CORRECTION),
```
Comment (argush3) [inline queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py:369]: CCC should be CC

File: queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py [hunk 5]
```diff
@@ -500,43 +512,43 @@ async def test_worker_director_name_and_address_change(app, session, mocker, tes
...
     'test_name',
     [
-        'ben_add_resolution_dates',
-        'ben_update_existing_resolution_dates',
-        'ben_update_with_new_resolution_dates',
-        'ben_delete_resolution_dates',
+        'bc_add_resolution_dates',
+        'bc_update_existing_resolution_dates',
+        'bc_update_with_new_resolution_dates',
+        'bc_delete_resolution_dates',
```
Comment (argush3) [inline queue_services/entity-filer/tests/unit/test_worker/test_correction_bcia.py:518]: can we update this test to test all the relevant entity types? i.e. BEN, BC, ULC, CC

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**